bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -29,211 +34,315 @@
             }
 
             String[] subkeys = key.split("\\.");
-            if (subkeys.length != 2) {
-                logger.debug("MQTT Broker property '{}' should have the format 'broker.propertykey'", key);
+            if (subkeys.length != 2 || StringUtils.isBlank(value)) {
+                logger.debug("MQTT Broker property '{}={}' should have the format 'broker.propertykey=value'", key,
+                        value);
                 continue;
             }
+            String brokername = subkeys[0].toLowerCase();
 
-            String value = (String) properties.get(key);
-            String name = subkeys[0].toLowerCase();
-            String property = subkeys[1];
-
-            if (StringUtils.isBlank(value)) {
-                logger.trace("Property is empty: {}", key);
-                continue;
-            } else {
-                logger.trace("Processing property: {} = {}", key, value);
+            Map<String, Object> brokerConfig = configPerBroker.get(brokername);
+            if (brokerConfig == null) {
+                brokerConfig = new TreeMap<>();
+                configPerBroker.put(brokername, brokerConfig);
+                brokerConfig.put("name", brokername);
             }
 
-            final MqttBrokerConnection conn = getConnection(name);
-
-            if (property.equals("url")) {
-                conn.setUrl(value);
-            } else if (property.equals("user")) {
-                conn.setUser(value);
-            } else if (property.equals("pwd")) {
-                conn.setPassword(value);
-            } else if (property.equals("qos")) {
-                conn.setQos(Integer.parseInt(value));
-            } else if (property.equals("retain")) {
-                conn.setRetain(Boolean.parseBoolean(value));
-            } else if (property.equals("async")) {
-                conn.setAsync(Boolean.parseBoolean(value));
-            } else if (property.equals("clientId")) {
-                conn.setClientId(value);
-            } else if (property.equals("lwt")) {
-                MqttWillAndTestament will = MqttWillAndTestament.fromString(value);
-                logger.debug("Setting last will: {}", will);
-                conn.setLastWill(will);
-            } else if (property.equals("keepAlive")) {
-                conn.setKeepAliveInterval(Integer.parseInt(value));
-            } else {
-                logger.warn("Unrecognized property: {}", key);
-            }
+            brokerConfig.put(subkeys[1], value);
         }
-        logger.info("MQTT Service initialization completed.");
 
-        startAllBrokerConnections();
+        return configPerBroker;
     }
 
     /**
-     * Start service.
+     * Create broker connections based on the service configuration.
      */
+    @Override
+    public void updated(Dictionary<String, ?> properties) {
+        // load broker configurations from configuration file
+        if (properties == null || properties.isEmpty()) {
+            return;
```

Comment:
Good point, to be sure I will disconnect and discard all existing, textual configured broker connections